lib/queuePopulator/KafkaLogConsumer/ListRecordStream.js

```diff
@@ -50,6 +50,29 @@ class ListRecordStream extends stream.Transform {
         }
     }
 
+    /**
+     * Get object metadata from oplog event
+     * @param {ChangeStreamDocument} changeStreamDocument changeStreamDocument
+     * @returns {string|undefined} stringified objectMd
+     */
+    _getObjectMd(changeStreamDocument) {
+        // Metadata values from updateDescription are used instead of
+        // fullDocument as they represent the exact changes that occured
+        // to the object. We only need this for the update operation.
+
+        // The fullDocument value contains a copy of the entire updated
+        // document at a point in time after the update, this can cause issues
+        // when the object gets deleted just after the update (only for updates).
+        const updateDescriptionMd = changeStreamDocument.updateDescription
+            && changeStreamDocument.updateDescription.updatedFields
+            && changeStreamDocument.updateDescription.updatedFields.value;
+        if (changeStreamDocument.operationType === 'update') {
+            return updateDescriptionMd && JSON.stringify(updateDescriptionMd);
+        }
+        const fullDocumentMd = changeStreamDocument.fullDocument && changeStreamDocument.fullDocument.value;
+        return fullDocumentMd && JSON.stringify(fullDocumentMd);
```

Comment:
just a question out of my curiosity

before it was `changeStreamDocument.fullDocument && JSON.stringify(changeStreamDocument.fullDocument.value),`

now we kind of changed it to sth like 
```
changeStreamDocument.fullDocument && 
changeStreamDocument.fullDocument.value &&
JSON.stringify(changeStreamDocument.fullDocument && 
changeStreamDocument.fullDocument.value)
```

what do we want to return here as value?

Reply:
Looking at it, there is a redundant check indeed, `JSON.stringify(undefined)` should equal `undefined`.
So this `return fullDocumentMd && JSON.stringify(fullDocumentMd);` should become `return JSON.stringify(fullDocumentMd);`

Reply:
ok cool! you already updated that 👍